### PR TITLE
Returns the entire MyriadConfiguration object as JSON from api/config

### PR DESCRIPTION
**Pull request by klucar**
GetConfigurationResponse is now just a subclass of MyriadConfiguration.
Added a copy constructor to MyriadConfiguration to enable this.
Standardized getters that return a boolean in MyriadConfiguration from
getBoolean to isBoolean. Removed duplicate getters.

### CODE REVIEW

File: src/main/java/com/ebay/myriad/configuration/MyriadConfiguration.java [hunk 1]
```diff
@@ -123,11 +123,35 @@
     @NotEmpty
     private Map<String, String> yarnEnvironment;
 
+
+    public MyriadConfiguration(){}
+
+    /*
+        Copy constructor used by rest api to map this config to GetConfigurationResponse
+     */
```
Comment (adam-mesos) [inline src/main/java/com/ebay/myriad/configuration/MyriadConfiguration.java:131]: Please use Javadoc style comments

File: src/main/java/com/ebay/myriad/configuration/MyriadConfiguration.java [hunk 2]
```diff
@@ -123,11 +123,35 @@
...
+        this.profiles = other.getProfiles();
+        this.rebalancer = other.isRebalancer();
+        this.nodemanager = other.getNodeManagerConfiguration();
+        this.executor = other.getMyriadExecutorConfiguration();
+        this.nativeLibrary = other.getNativeLibrary();
+        this.zkServers = other.getZkServers();
+        this.zkTimeout = other.getZkTimeout();
+        this.restApiPort = other.getRestApiPort();
+        this.yarnEnvironment = other.getYarnEnvironment();
+    }
```
Comment (adam-mesos) [inline src/main/java/com/ebay/myriad/configuration/MyriadConfiguration.java:147]: Please also include mesosAuthenticationPrincipal and mesosAuthenticationSecretFilename and anything else added in the meantime.

**Comment by adam-mesos** [comment]
Looks good, but probably needs a rebase (new authentication parameters).
Is there an issue that this PR relates to (or is cleanup for)? If so, please link it here.

**Comment by klucar** [comment]
A lot probably needs a rebase at this point. I think I'd rather wait for
more PRs to be incorporated before I start updating PRs.

As for this specifically, I just noticed that there were essentially two
configuration objects. One that was the actual configuration and one that
was published by the api endpoint. Making them the same thing seemed to
make sense to me.

Most likely happened when I was working on

https://github.com/mesos/myriad/issues/8
or
https://github.com/mesos/myriad/issues/54

**Comment by smarella** [comment]
I just tested this PR locally in my dev env. The API worked well. I was pleasantly surprised that the API also returned the new authentication params although the copy constructor didn't copy them (as yet). Digging further, I realized that  MyriadConfiguration object is actually a singleton in Myriad. MyriadModule.java has the following:

``` java
           bind(MyriadConfiguration.class).toInstance(cfg);
```

Which means, we may not really need an intermediate GetConfigurationResponse.java class that extends MyriadConfiguration. I further refactored the code, removed GetConfigurationResponse entirely and instead returned MyriadConfiguration as response to getConfig() method in ConfigurationResource.java. Like so:

``` java
@Path("/api/config")
@Produces(MediaType.APPLICATION_JSON)
public class ConfigurationResource {
    private MyriadConfiguration cfg;

    @Inject
    public ConfigurationResource(MyriadConfiguration cfg) {
        this.cfg = cfg;
    }

    @Timed
    @GET
    public MyriadConfiguration getConfig() throws InterruptedException {
        return this.cfg; // <===== directly return the MyriadConfiguration 
    }
}
```

This worked exactly the same way as it previously did with GetConfigurationResponse. Sample output from my testing (please note that mesosAuthenticationPrincipal and mesosAuthenticationSecretFilename properties are also part of the output):

``` json
{
    "mesosMaster": "10.10.30.131:5050",
    "checkpoint": false,
    "frameworkFailoverTimeout": 43200000,
    "frameworkName": "MyriadAlpha",
    "frameworkRole": "",
    "profiles": {
        "small": {
            "cpu": "1",
            "mem": "1100"
        },
        "medium": {
            "cpu": "2",
            "mem": "2048"
        },
        "large": {
            "cpu": "4",
            "mem": "4096"
        }
    },
    "rebalancer": false,
    "nativeLibrary": "/usr/local/lib/libmesos.so",
    "zkServers": "10.10.30.131:5181",
    "zkTimeout": 20000,
    "restApiPort": 8192,
    "yarnEnvironment": {
        "YARN_HOME": "/opt/mapr/hadoop/hadoop-2.5.1/",
        "YARN_NODEMANAGER_OPTS": "-Dyarn.nodemanager.resources.io-spindles=4.0 -Dyarn.resourcemanager.hostname=10.10.30.132"
    },
    "mesosAuthenticationPrincipal": "",
    "mesosAuthenticationSecretFilename": "",
    "nodeManagerConfiguration": {
        "jvmMaxMemoryMB": {
            "present": true
        },
        "user": {
            "present": true
        },
        "cpus": {
            "present": true
        },
        "jvmOpts": {
            "present": false
        },
        "cgroups": {
            "present": true
        }
    },
    "myriadExecutorConfiguration": {
        "jvmMaxMemoryMB": {
            "present": true
        },
        "path": "file:///root/myriad-executor-0.0.1.jar"
    }
}
```

**Comment by klucar** [comment]
Awesome. +1 for less code and not keeping that copy constructor up to date.

**Comment by smarella** [comment]
Thanks Jim. I've merged your changes locally and committed the PR: https://github.com/mesos/myriad/commit/eebb800dd244ec41e28c8f39c68aaf5de861c67b

Updated the API doc with the new API.

**Pull request closed without merging**